core/src/main/scala/org/apache/spark/scheduler/cluster/SparkDeploySchedulerBackend.scala

```diff
@@ -78,7 +78,11 @@ private[spark] class SparkDeploySchedulerBackend(
     val appUIAddress = sc.ui.map(_.appUIAddress).getOrElse("")
     val appDesc = new ApplicationDescription(sc.appName, maxCores, sc.executorMemory, command,
       appUIAddress, sc.eventLogDir)
-
+    appDesc.coreNumPerTask = conf.getInt("spark.task.cpus", 1)
```

Comment:
I believe use an `val ` to represent `spark.task.cpus` is better, like `val maxCores = conf.getOption("spark.cores.max").map(_.toInt)`.